src/components/Verse/VerseSettingsMenu.tsx

```diff
@@ -0,0 +1,54 @@
+import { useState, useEffect } from 'react';
+import styled from 'styled-components';
+import Verse from '../../../types/Verse';
+import VerseSettingsMenuItem from './VerseSettingsMenuItem';
+import CopyIcon from '../../../public/icons/copy.svg';
+
+interface Props {
+  verse: Verse;
+}
+
+const VerseSettingsMenu: React.FC<Props> = ({ verse }) => {
+  const [copied, setCopied] = useState(false);
+
+  useEffect(() => {
+    let timeoutId: NodeJS.Timeout;
+    // if the user has just copied the text, we should change the text back to Copy after 3 seconds.
+    if (copied === true) {
+      timeoutId = setTimeout(() => setCopied(false), 3 * 1000);
+    }
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, [copied]);
+
+  const onCopyClicked = () => {
+    navigator.clipboard
+      .writeText(
+        // remove the last element which is the verse's number then join all of them into 1 string
+        verse.words
```

Comment:
Copying v2, and v3 glyph codes don't make any sense. We can fetch text uthmani for each ayah using `fields` query string in [makeVersesUrl](https://github.com/quran/quran.com-frontend-next/blob/4ba85e5cdb7be1e2d8deeb2d9141d042808c26ec/src/utils/apiPaths.ts#L11)

Then copy Uthmani text of ayah.

Reply:
JAK for pointing that out. I assumed that the user would want to copy the verse in the current selected Quran font. I will include `text_uthmani` under `fields` when calling the API.